### DISCUSSION TITLE
Watts measurements.

### DISCUSSION THREAD

**Discussion by rowlest** [JGAguado/Smart_Powermeter#26] [General]
Hi got your system up and running! Very fun.
Just wondering if voltage is measured at all so the watts can be measured for each circuit accurately ?

Not just w = 230 * a 

Also with your base yaml be nice if it was full fleshed out with a complete setup eg
 LED power on boot
Total kw for 6 probes not 3
Watts returned not just amps for each circuit

Also I couldn't flash via USB had to be done via serial but all working OTA/USB once flashed.

**Comment by JGAguado** [comment]
Hi @rowlest !
Good to know that you liked it and got it up an running so fast! I believe you are one of the first trying the V2R2 batch 😉

With this board, as with the V2R1, only current can be measured, and power (aparent power) computed afterwards with the formula.
Thanks for the hint with the LED for the base YAML, the rest of the file should serve just as simple example, after all each user adapts it later to their own needs. 

Can you tell me more about the issue with the USB for flashing it? Did you follow the sequence of pressing the *flash* button while resetting the board?

**Reply by rowlest** [reply]
Hi @JGAguado ,

Thank you for moving to discussions, my bad. 
Ok no worries, was wondering. Thank you for your input. 

With the USB it entered flashing mode but wouldnt accept the base esphome flashing (sorry I didnt keep the logs). I've done it with many other USB devices not sure what was going one. But first flash with serial worked fine. 

Also one thing that isnt mentioned when wiring the AC input is which side active and neutral go. I looked at one of the pictures in the switch panel for reference. But no markings in the docs, board, or pins.

**Reply by JGAguado** [reply]
Dear @rowlest 

According to the HLK-2M05 datasheet (https://www.hlktech.com/en/Goods-39.html) it doesn't seem to matter the AC input layout.

Regarding the USB, did you flashed it from the ESPHome Add-on? You remember if after entering into flash mode, you could see the port `ESP32-S2` or nothing at all? 

Thanks and best regards,

Jon

**Reply by mi-hol** [reply]
> With this board, as with the V2R1, only current can be measured, and power (aparent power) computed afterwards with the formula.

to compute power in Watts correctly, the current voltage needs to be known too!

Fact: during times of high PV output, grid voltage goes up to 251V in my local grid
Assuming voltage to be 230V introduces an error of nearly 10%

949W / 251V = 3,78A
3,78A * 230V = 869,6 W

Delta = 8.7%

Could perhaps a new hardware revision resolve this issue?

![image](https://github.com/JGAguado/Smart_Powermeter/assets/22799428/27e7466c-def9-4003-83ac-6f1878524df9)

![image](https://github.com/JGAguado/Smart_Powermeter/assets/22799428/0ddb5984-a6d8-492f-be1f-e7c64443b60a)

**Reply by JGAguado** [reply]
That's correct, the current revision only provide apparent power, which is not necessary the same as the real one. 

And yes, I'm working on a revision capable of measuring also voltage, but it might take me some time to get it ready 😉😉

**Reply by mi-hol** [reply]
> working on a revision capable of measuring also voltage

I'd be happy to test a prototype :)

**Reply by lucazeta1975** [reply]
Considering that the Smart Powermeter can also be also powered at 5V, in the meantime, a third-party voltage sensor could be used, and then cross-reference the Ampere * Voltage data by creating template-type sensors in Home Assistant.

This is not an all-in-one solution, but it could be good for inconsistent voltage AC mains

**Reply by Saentist** [reply]
Instead to use fixed voltage in code can be taken from other device 

```yaml
# Example configuration entry
sensor:
  - platform: homeassistant
    name: "Voltage Sensor From Home Assistant"
    entity_id: sensor.voltage_sensor
```
